pkg/client/client.go

```diff
@@ -301,9 +302,15 @@ func (tc *TelemetryClient) Register() (err error) {
 }
 
 func (tc *TelemetryClient) Generate(telemetry types.TelemetryType, content []byte, tags types.Tags) error {
+	// Enforce size limits
+	_, err := telemetrylib.NewTelemetryDataLimits(content)
```

Comment:
A NewSomething() style routine should conform to the Go new() coding pattern, i.e. you call the NewSomething() routine to option a new object that has minimally initialised, ready to call methods on. The arguments to such a NewSomething() call should be related to minimally initialising the created object, e.g. dimensions for initilising contained objects and structures.

The pattern you have here is more of a top-level module helper function style, which internally might get a new object instance, possibly using a NewSomething() routine, and use it to perform the desired action, without requiring the caller to do any intermediate setup work.

I would rename your NewTelemetryDataLimits() helper to CheckTelemetryDataLimits() and there is no need for it to return the TelemetryDataLimits() pointer any more.

But for future potential use cases it would be useful to implement a NewTelemetryDataLimits() helper that returns a ready to use TelemetryDataLimits object pointer.